### PR TITLE
GEN-2382 - feat: added support for content structured data

### PR DESCRIPTION
**Pull request by guilhermespopolin**
## Describe your changes

* Added support for content structured data (Article, NewsArticle, BlogPosting and ProfilePage).

image

## Justify why they are needed

Requested by Peter

### CODE REVIEW

**Comment by guilhermespopolin** [comment]
* **#4409** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4409?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @guilhermespopolin and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/store/src/app/[locale]/[[...slug]]/ContentCmsPage.tsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import { type RoutingLocale } from '@/utils/l10n/types'
+import { BlogStoryContainer } from './BlogStoryContainer'
+import { ContentTypeStructuredData } from './ContentTypeStructuredData'
+
+type Props = {
+  locale: RoutingLocale
+  story: PageStory
+}
+
+export function ContentCmsPage({ locale, story }: Props) {
```
Comment (alebedev) [inline apps/store/src/app/[locale]/[[...slug]]/ContentCmsPage.tsx:13]: This is a great name!

File: apps/store/src/app/[locale]/[[...slug]]/ContentTypeStructuredData.tsx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+      }}
+    />
+  )
+}
+
+function ProfilePageStructuredData({ story }: { story: PageStory }) {
+  const { image, datePublished, dateModified, authorName } = story.content
+
+  const mainEntity =
+    authorName || image
```
Comment (alebedev) [inline apps/store/src/app/[locale]/[[...slug]]/ContentTypeStructuredData.tsx:55]: Are we supposed to always have authorName?  No need to check image, if so

File: apps/store/src/app/[locale]/[[...slug]]/ContentTypeStructuredData.tsx [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+          '@type': 'Person',
+          name: authorName,
+          image: image?.filename,
+        }
+      : undefined
+
+  return (
+    <script
+      key="article-structured-data"
+      // TODO: remove this
```
Comment (alebedev) [inline apps/store/src/app/[locale]/[[...slug]]/ContentTypeStructuredData.tsx:66]: This PR?

**Review by alebedev — APPROVED** [review]
Looks good

File: apps/store/src/app/[locale]/[[...slug]]/ContentTypeStructuredData.tsx (hunk 3 above)
Comment (guilhermespopolin) [reply apps/store/src/app/[locale]/[[...slug]]/ContentTypeStructuredData.tsx:66]: 😅

File: apps/store/src/app/[locale]/[[...slug]]/ContentTypeStructuredData.tsx (hunk 2 above)
Comment (guilhermespopolin) [reply apps/store/src/app/[locale]/[[...slug]]/ContentTypeStructuredData.tsx:55]: No, all of them are optional :)

**Comment by guilhermespopolin** [comment]
### Merge activity

* **Jun 27, 3:16 AM EDT**: @guilhermespopolin started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4409).

File: apps/store/src/app/[locale]/[[...slug]]/ContentTypeStructuredData.tsx [hunk 4]
```diff
@@ -0,0 +1,82 @@
+import { formatISO } from 'date-fns'
+import { type PageStory } from '@/services/storyblok/storyblok'
+
+export function ContentTypeStructuredData({ story }: { story: PageStory }) {
+  const { type } = story.content
+
+  if (!type) return null
+
+  // Just being extra careful here and covering the case where we get a uncodversed type
```
Comment (gustaveen) [inline apps/store/src/app/[locale]/[[...slug]]/ContentTypeStructuredData.tsx:9]:
```suggestion
  // Just being extra careful here and covering the case where we get a uncovered type
```
typo?

File: apps/store/src/app/[locale]/[[...slug]]/ContentTypeStructuredData.tsx [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+
+function ProfilePageStructuredData({ story }: { story: PageStory }) {
+  const { image, datePublished, dateModified, authorName } = story.content
+
+  const mainEntity =
+    authorName || image
+      ? {
+          '@type': 'Person',
+          name: authorName,
+          image: image?.filename,
```
Comment (gustaveen) [inline apps/store/src/app/[locale]/[[...slug]]/ContentTypeStructuredData.tsx:57]: nit: Just wondering, is that supposed to be an image of the author? If so, should we name it authorImage? 

And will we we ever upload author image?

**Review by gustaveen — APPROVED** [review]
Good job! 🥳

**Pull request merged**